diff --git a/Users/maxim/Downloads/LSP/plugin/core/views.py b/Users/maxim/Library/Application Support/Sublime Text/Packages/LSP/plugin/core/views.py
index 0319305..b5a72a2 100644
--- a/Users/maxim/Downloads/LSP/plugin/core/views.py
+++ b/Users/maxim/Library/Application Support/Sublime Text/Packages/LSP/plugin/core/views.py	
@@ -1056,6 +1056,9 @@ def format_completion(
     insert_replace_support_html = get_insert_replace_support_html(item)
     if insert_replace_support_html:
         details.append(insert_replace_support_html)
+    if kind == KIND_FUNCTION or kind == KIND_METHOD:
+      _, sep, after = lsp_label.partition('(')
+      trigger += sep + after
 
     completion = sublime.CompletionItem.command_completion(
         trigger=trigger,
